src/Microsoft.AspNet.OData.Shared/DeltaSetOfT.cs

```diff
@@ -0,0 +1,357 @@
+
+                        deltaSet.Add(changedObj);
+                    }
+                }
+                catch(Exception ex)
+                {
+                    //For handling the failed operations.
+                    IDeltaSetItem changedObject = HandleFailedOperation(changedObj, operation, original, ex.Message);                    
+                    deltaSet.Add(changedObject);
+                }
+            }
+
+            return deltaSet;
+        }
+
+
+        private DeltaSet<TStructuralType> CreateDetlaSet()
+        {
+            Type type = typeof(DeltaSet<>).MakeGenericType(_clrType);
+
+            DeltaSet<TStructuralType> deltaSet = Activator.CreateInstance(type, _keys) as DeltaSet<TStructuralType>;
+            return deltaSet;
+        }
+
+        private IDeltaSetItem HandleFailedOperation(Delta<TStructuralType> changedObj, DataModificationOperationKind operation, TStructuralType originalObj, string errorMessage)
+        {
+            IDeltaSetItem deltaSetItem = null;
+            DataModificationExceptionType dataModificationExceptionType = new DataModificationExceptionType(operation);
+            dataModificationExceptionType.MessageType = new MessageType();
+            dataModificationExceptionType.MessageType.Message = errorMessage;
+
+            // This handles the Data Modification exception. This adds Core.DataModificationException annotation and also copy other instance annotations.
+            //The failed operation will be based on the protocol
+            switch (operation)
+            {
+                case DataModificationOperationKind.Update:
+                    deltaSetItem = changedObj;
+                    break;
+                case DataModificationOperationKind.Insert:
+                    {
+                        deltaSetItem = CreateDeletedEntityForFailedOperation(changedObj);
+
+                        break;
+                    }
+                case DataModificationOperationKind.Delete:
+                    {
+                        deltaSetItem = CreateEntityObjectforFailedOperation(changedObj, originalObj);                        
+                        break;
+                    }
+            }
+
+
+            deltaSetItem.TransientInstanceAnnotationContainer = changedObj.TransientInstanceAnnotationContainer;
+            deltaSetItem.TransientInstanceAnnotationContainer.AddResourceAnnotation("Core.DataModificationException", dataModificationExceptionType);
+
+            Contract.Assert(deltaSetItem != null);
+
+            return deltaSetItem;
+        }
+
+        private IDeltaSetItem CreateEntityObjectforFailedOperation(Delta<TStructuralType> changedObj, TStructuralType originalObj)
+        {
+            Type type = typeof(Delta<>).MakeGenericType(_clrType);
+
+            Delta<TStructuralType> deltaObject = Activator.CreateInstance(type, _clrType, _clrType.GetProperties().Select(x=>x.Name), null,
```

Comment:
>_clrType.GetProperties().Select(x=>x.Name) [](start = 90, length = 42)

do we need to specify this? can't we pass null and let InitializeProperties in Delta<T> take care of this?  I think the logic there is better than the logic here (i.e., makes sure they are settable public instance properties, etc...)